beacon-chain/db/kv/blocks.go

```diff
@@ -262,28 +262,34 @@ func (s *Store) SaveBlocks(ctx context.Context, blocks []block.SignedBeaconBlock
 	ctx, span := trace.StartSpan(ctx, "BeaconDB.SaveBlocks")
 	defer span.End()
 
+	blockRoots := make([][]byte, len(blocks))
+	encodedBlocks := make([][]byte, len(blocks))
+	indicesForBlocks := make([]map[string][]byte, len(blocks))
+	for i, blk := range blocks {
+		blockRoot, err := blk.Block().HashTreeRoot()
+		if err != nil {
+			return err
+		}
+		enc, err := marshalBlock(ctx, blk)
+		if err != nil {
+			return err
+		}
+		blockRoots[i] = blockRoot[:]
+		encodedBlocks[i] = enc
+		indicesByBucket := createBlockIndicesFromBlock(ctx, blk.Block())
+		indicesForBlocks[i] = indicesByBucket
```

Comment:
My only concern here is that previously, we would be doing a check in the DB to see if the block is an existing block, if so, then we continue. Now, we are performing these operations always. Perhaps this is worth the trade-off?

Reply:
I think it is fine, we do not expect to save duplicate blocks often/at-all .